features/GEM003_Unique-representation-identifier.feature

```diff
@@ -0,0 +1,11 @@
+@informal-proposition
+@GEM
+Feature: GEM003 - Unique Representation Identifer
+
+  Scenario: No Shape Representation identifier shall be used twice within the product representation of an element 
+
+      Given An IfcProductDefinitionShape
+        And The element has more than 1 instance(s) of IfcShapeRepresentation
+
+        Then Each instance of IfcShapeRepresentation has a unique value for the attribute RepresentationIdentifier
```

Comment:
I would question here whether it's not better to use the attribute name in gherkin as opposed to the type of attribute value. It save a lookup table in python but also works better if there are multiple attributes of the same type.

Reply:
What is lacking here is really composability. Point for discussion. How would you feel about this:

```
Given  An IfcProductDefinitionShape
And    Its values for attribute Representations
And    Its values for attribute RepresentationIdentifier

Then   The values should be unique
```

I'm not convinced whether this is a good idea. But you can see the kind of composability and reuse it gives. But it's confusing that the Given statement becomes so order-dependent and so dramatically alters the set of instances/values that later statements operate on.